src/views/AuthoritiesSearch/AuthoritiesSearch.js

```diff
@@ -164,15 +165,29 @@ const AuthoritiesSearch = ({
   const [storedFilterPaneVisibility] = useLocalStorage(filterPaneVisibilityKey, true);
   const [isFilterPaneVisible, setIsFilterPaneVisible] = useState(storedFilterPaneVisibility);
   const [selectedRows, setSelectedRows] = useState({});
+  const [selectAll, setSelectAll] = useState(false);
+
+  const selectedRowsCount = useMemo(() => (Object.keys(selectedRows).length), [selectedRows]);
+  const uniqueAuthoritiesCount = useMemo(() => {
+    // determine count of unique ids in authorities array.
+    // this is needed to check or uncheck "Select all" checkbox in header when all rows are explicitly
+    // checked or unchecked.
+    const filteredAuthorities = [];
```

Comment:
This can be done with a two lines: 
```
const filteredAuthorities = authorities.map(authority => authority.id).filter(id => !!id);
const uniqueAuthoritiesCount = new Set(filteredAuthorities).size;
```